ionoscloud/data_source_k8s_clusters.go

```diff
@@ -0,0 +1,247 @@
+		clusterConfigProperties, err := K8sClusterConfigProperties(clusterConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		// node pools
+		clusterNodePools, apiResponse, err := client.KubernetesApi.K8sNodepoolsGet(ctx, *cluster.Id).Execute()
+		logApiRequestTime(apiResponse)
+		if err != nil {
+			return nil, fmt.Errorf("an error occurred while fetching the kubernetes cluster node pools for cluster with ID %s: %w", *cluster.Id, err)
+		}
+		if clusterNodePools.Items != nil && len(*clusterNodePools.Items) > 0 {
+			var nodePools []any
+			for _, nodePool := range *clusterNodePools.Items {
+				nodePools = append(nodePools, *nodePool.Id)
+			}
+			clusterProperties["node_pools"] = nodePools
+		}
+
+		clusterProperties = mergeMaps(clusterProperties, clusterConfigProperties)
+	}
+
+	return clusterProperties, nil
+}
+
+// K8sClusterConfigProperties returns a map with additional properties parsed from the kubeconfig of a KubernetesCluster
+func K8sClusterConfigProperties(clusterConfig string) (map[string]any, error) {
+	kubeConfig := KubeConfig{}
+	if err := yaml.Unmarshal([]byte(clusterConfig), &kubeConfig); err != nil {
+		return nil, err
+	}
+
+	clusterProperties := make(map[string]any)
+	clusterConfigProperties := make(map[string]any)
+
+	clusterConfigProperties["api_version"] = kubeConfig.ApiVersion
+	clusterConfigProperties["current_context"] = kubeConfig.CurrentContext
+	clusterConfigProperties["kind"] = kubeConfig.Kind
+
+	var decodedCert string
+	clusters := make([]map[string]any, len(kubeConfig.Clusters))
+	for i, c := range kubeConfig.Clusters {
+		caCert := make([]byte, base64.StdEncoding.DecodedLen(len(c.Cluster.CaData)))
+		if _, err := base64.StdEncoding.Decode(caCert, []byte(c.Cluster.CaData)); err != nil {
+			return nil, err
+		}
+		decodedCert = string(caCert)
+		clusters[i] = map[string]any{"name": c.Name, "cluster": map[string]string{"server": c.Cluster.Server, "certificate_authority_data": decodedCert}}
+	}
+	clusterConfigProperties["clusters"] = clusters
+
+	contexts := make([]map[string]any, len(kubeConfig.Contexts))
+	for i, contextVal := range kubeConfig.Contexts {
+		contexts[i] = map[string]any{
+			"name": contextVal.Name,
+			"context": map[string]string{
+				"cluster": contextVal.Context.Cluster,
+				"user":    contextVal.Context.User,
+			},
+		}
+	}
+	clusterConfigProperties["contexts"] = contexts
+
+	users := make([]map[string]any, len(kubeConfig.Users))
+	userTokens := make(map[string]string)
+	for i, user := range kubeConfig.Users {
+		users[i] = map[string]any{
+			"name": user.Name,
+			"user": map[string]any{
+				"token": user.User.Token,
+			},
+		}
+		userTokens[user.Name] = user.User.Token
+	}
+	clusterConfigProperties["users"] = users
+
+	clusterProperties["config"] = []map[string]any{clusterConfigProperties}
+	clusterProperties["ca_crt"] = decodedCert
+	clusterProperties["user_tokens"] = userTokens
+
+	return clusterProperties, nil
+}
+
+func mergeMaps(maps ...map[string]any) map[string]any {
```

Comment:
should his be in utils? Doesn't seem to be k8s related

Reply:
agreed, moved to utils and added some comments ([b9a7d92](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/519/commits/b9a7d92cfbcc64f79aa318a0eeab6fc491584516))